### PR TITLE
datasets pages

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Add the LLM Prompt Datasets pages to the Evidence CMS, with the requisite backend functionality to make it possible to upload and view these datasets.

## WHY
This is Part 1 of Project 2 to pull Brendan's work into the main Evidence CMS.

## HOW
Mostly React and CSS following existing paradigms on the frontend. In order to submit CSV data from the frontend, I had to write a custom fetch implementation, and split how we handle json vs html requests on the backend. I added a TODO to clean this up at the end of this project.

Two extras: 
1. I made an accessibility update to our `TextArea` component that changed a couple of additional snapshot files, if you're wondering what those are doing in here.
2. I realized the way I was using `split` on the prompt to get the stem for the `StemVault` sometimes split too early, so the logic there has changed slightly to use `rpartition`.

### Screenshots
Screenshot 2024-10-23 at 4 17 35 PM
Screenshot 2024-10-23 at 4 20 52 PM
Screenshot 2024-10-23 at 4 52 49 PM
Screenshot 2024-10-23 at 4 53 01 PM

### Notion Card Links
https://www.notion.so/quill/LLM-Prompts-Datasets-Pages-126d42e6f94180df8eb2e540fc188612#126d42e6f94180f9ad79c5130a2b19b7

### What have you done to QA this feature?
Tested dataset upload both locally and on staging, including versions with errors.

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

File: services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb [hunk 1]
```diff
@@ -36,9 +45,20 @@ def show
           end
         end
 
+        private def create_dataset_from_file_for_json_request
+          @dataset = stem_vault.datasets.new(dataset_params)
```
Comment (brendanshean) [inline services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb]: You'll need to add task_type into this initializer.  

There's no way you would have known this as the change was just merged in from my RAG stuff.  (see line 38 above)

Something like:
`@dataset = stem_vault.datasets.new(dataset_params.merge(task_type: Dataset::GENERATIVE))`

File: services/QuillLMS/engines/evidence/app/services/evidence/gen_ai_record_builder.rb [hunk 2]
```diff
@@ -44,7 +44,9 @@ def create_or_update_stem_vault(prompt, gen_ai_activity)
         conjunction: prompt.conjunction
       )
 
-      stem_vault.stem = prompt.text.split(prompt.conjunction).first.strip
+      partitioned_stem = prompt.text.rpartition(prompt.conjunction).first
```
Comment (brendanshean) [inline services/QuillLMS/engines/evidence/app/services/evidence/gen_ai_record_builder.rb:47]: Cool use of `rpartition`

**Review by brendanshean — APPROVED** [review]
Wow.  This looks awesome.   Great to see this taking shape!

File: services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptTrials/stemVaultSection.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    name: 'Trials',
+    attribute: 'trialsCount',
+    width: '30px',
+    headerClassName: 'center-content',
+    rowSectionClassName: 'center-content'
+  },
+  {
+    name: 'Access',
+    attribute: 'viewButton',
+    headerClassName: 'center-content',
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptTrials/stemVaultSection.tsx:59]: should we just make this into a const?

**Review by eadams17 — APPROVED** [review]
This looks great! Excited to see this moving along

**Pull request merged**